jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanReader.java

```diff
@@ -161,6 +162,7 @@ public Object read(JSONReader r, JsonParser p) throws IOException
                     // [jackson-jr#148] Record deser support (2.18)
                     if (_isRecordType) {
                         final List<Object> values = new ArrayList<>();
+                        final List<Object> props = new ArrayList<>();
```

Comment:
only if we can store props and its value as a pair in a single list, it would've been simpler.. 🙂